src/main/webapp/js/index.js

```diff
@@ -12,8 +12,92 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/* Date range picker handler:
- * https://www.daterangepicker.com/
+
+/** Fetches receipts from the server and adds them to the DOM. */
+async function searchReceipts() {
+  const label = document.getElementById('search-input').value;
+  const response = await fetch(`/search-receipts?label=${label}`);
+  const receipts = await response.json();
+
+  clearExistingDisplay();
+  displayReceipts(label, receipts);
+}
+
+/**
+ * Clears out receipts display including old receipts and error messages.
+ */
+function clearExistingDisplay() {
+  const elements = document.querySelectorAll(
+      '.col-md-6, .col-md-12.text-center.error-message');
```

Comment:
This method is a lot cleaner now, but referencing CSS classes can make this brittle (e.g., if you decide to change the style, someone needs to make sure to update this reference in the javascript. I do think you could declare the templates outside of the `receipts-display` div element and then you would be able to just do `document.getElementById('receipts-display').innerHtml = '';` to reset things.

Reply:
Oh I see what you mean now. I didn't know templates could be placed anywhere in the html and then attached where you want them. That would've saved me a lot of headache